FrontEnd/src/pages/Cupos/formularioCupos.jsx

```diff
@@ -0,0 +1,261 @@
+import { useEffect, useState } from "react";
+import { useLocation, useParams, useNavigate } from "react-router-dom";
+import { useCupos } from "../../context/Cupo/proveedorCupo.jsx";
+import { useUsuario } from "../../context/Usuario/proveedorUsuario.jsx";
+
+const FormularioCupos = () => {
+  const [values, setValues] = useState({
+    horario: "",
+    estado: "active", // Estado por defecto para un nuevo cupo
+    cupo: "",
+    dniInstructor: "",
+    diaSemana: "",
+  });
+
+  const { obtenerProfesionales, profesionales } = useUsuario(); 
+
+  const [showModal, setShowModal] = useState(false);
+  const [modalMessage, setModalMessage] = useState("");
+  const [isSuccess, setIsSuccess] = useState(false);
+
+  const location = useLocation();
+  const { idEsquema, diaSemana } = useParams();
+  const navigate = useNavigate();
+
+  const { crearCupo, actualizarCupo, cargarCupoXid } = useCupos();
+
+  const isNewRoute = location.pathname.includes("/new");
+  const diaPreseleccionado = diaSemana;
+
+  const dayMapping = {
+    domingo: 0,
```

Comment:
mejor usar una lib o el mismo date para esto https://stackoverflow.com/questions/24998624/day-name-from-date-in-js

Reply:
lo pasamos a manejar con date